src/RoleObjects/SpinButtonData.test.js

```diff
@@ -0,0 +1,190 @@
+import * as createjs from 'createjs-module';
+import ReactTestUtils from 'react-dom/test-utils';
+import KeyCodes from 'keycodes-enum';
+import AccessibilityModule from '../index';
+import { parentEl, stage, container } from '../__jestSharedSetup';
+
+describe('SpinButtonData', () => {
+  describe('register role', () => {
+    let cjsSpin;
+    let spinEl;
+    const accessibleOptions = {};
+
+    beforeEach(() => {
+      cjsSpin = new createjs.Shape();
+      accessibleOptions.enableKeyEvents = false;
+      accessibleOptions.value = 10;
+      accessibleOptions.min = 5;
+      accessibleOptions.max = 50;
+      accessibleOptions.readOnly = false;
+      accessibleOptions.required = false;
+      AccessibilityModule.register({
+        displayObject: cjsSpin,
+        parent: container,
+        role: AccessibilityModule.ROLES.SPINBUTTON,
+        accessibleOptions,
+      });
+
+      stage.accessibilityTranslator.update();
+      spinEl = parentEl.querySelector(`#${cjsSpin.accessible.domId}`);
+    });
+
+    describe('rendering', () => {
+      it('creates spin Button element', () => {
+        expect(spinEl).not.toBeUndefined();
+      });
+    });
+
+    describe('"addChild" and "addChildAt"', () => {
+      it('thorws Error if child is added', () => {
+        const childObj = new createjs.Shape();
+        expect(() => {
+          cjsSpin.accessible.addChild(childObj);
+        }).toThrowError(/spinbutton cannot have children/);
+        expect(() => {
+          cjsSpin.accessible.addChildAt(childObj, 0);
+        }).toThrowError(/spinbutton cannot have children/);
+      });
+    });
+
+    describe('Accessible getters and setters methods', () => {
+      it('can get and set "enableKeyEvent" property', () => {
+        expect(cjsSpin.accessible.enableKeyEvents).toBe(false);
+
+        cjsSpin.accessible.enableKeyEvents = true;
+        expect(cjsSpin.accessible.enableKeyEvents).toBe(true);
+      });
+
+      it('can get and set "value" property', () => {
+        expect(cjsSpin.accessible.value).toBe(accessibleOptions.value);
+
+        cjsSpin.accessible.value = 25;
+        stage.accessibilityTranslator.update();
+        expect(cjsSpin.accessible.value).toBe(25);
+        expect(+spinEl.getAttribute('value')).toBe(25);
+      });
+
+      it('can get and set "min" and "max" property', () => {
+        expect(cjsSpin.accessible.min).toBe(accessibleOptions.min);
+        expect(cjsSpin.accessible.max).toBe(accessibleOptions.max);
+
+        cjsSpin.accessible.min = 0;
+        cjsSpin.accessible.max = 100;
+        stage.accessibilityTranslator.update();
+        expect(cjsSpin.accessible.min).toBe(0);
+        expect(cjsSpin.accessible.max).toBe(100);
+        expect(+spinEl.getAttribute('min')).toBe(0);
+        expect(+spinEl.getAttribute('max')).toBe(100);
+      });
```

Comment:
- please separate these so that each assertion passes or fails independently - can tie them together with a describe and have `min` be one assertion and `max` be another